packages/components/src/tools-panel/test/index.tsx

```diff
@@ -21,49 +22,49 @@ const defaultProps = {
 };
 
 // Default props for an enabled control to be rendered within panel.
+let controlValue: boolean | undefined = true;
 const controlProps = {
-	attributes: { value: true },
 	hasValue: jest.fn().mockImplementation( () => {
-		return !! controlProps.attributes.value;
+		return !! controlValue;
 	} ),
 	label: 'Example',
 	onDeselect: jest.fn().mockImplementation( () => {
-		controlProps.attributes.value = undefined;
+		controlValue = undefined;
 	} ),
 	onSelect: jest.fn(),
 };
 
 // Default props without a value for an alternate control to be rendered within
 // the panel.
+let altControlValue: boolean | undefined = false;
 const altControlProps = {
-	attributes: { value: false },
 	hasValue: jest.fn().mockImplementation( () => {
-		return !! altControlProps.attributes.value;
+		return !! altControlValue;
 	} ),
 	label: 'Alt',
 	onDeselect: jest.fn(),
 	onSelect: jest.fn(),
 };
 
 // Default props for wrapped or grouped panel items.
+let nestedControlValue: boolean | undefined = true;
```

Comment:
Nit: seems like this type gets repetitive and we could have an inline `ControlValue` type.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/48275/commits/62fabcfb1791fe639b736abe7e43cb8bca33199b